cli/src/pcluster/validators/networking_validators.py

```diff
@@ -47,6 +47,15 @@ def _validate(self, subnet_ids: List[str]):
                         FailureLevel.ERROR,
                     )
 
+            # warn if subnets are not in the same AZ
+            azs = list(dict.fromkeys([subnet.get("AvailabilityZone") for subnet in subnets]))
```

Comment:
this can be simply replaced by a set rather than doing list -> dict -> list conversion.

Also differently from the schema level validator this is also including the head node subnet which becomes a breaking change if, as suggested, we transform the failure below into an error.

Reply:
Gotcha so I transformed this into a set however I'd like to keep it as a warning so users don't have to set an extra flag to get it working